samples/renderer/src/winmain.cpp

```diff
@@ -0,0 +1,211 @@
+
+            auto brick_wall_tex = rcbe::visual::make_tex_ptr(
+                "external/brick_wall_texture/file/brick_wall_texture.tga", rdmn::parse::tga::parse);
+            auto awesome_face_tex = rcbe::visual::make_tex_ptr(
+                "external/awesomeface_texture/file/awesomeface_texture.tga", rdmn::parse::tga::parse);
+
+            rcbe::visual::VisualTextureSet vts {brick_wall_tex, awesome_face_tex};
+
+            std::vector<rcbe::rendering::Material::TextureArguments> texture_args;
+            texture_args.reserve(2);
+            texture_args.push_back({brick_wall_tex, raster_tex_conf});
+            texture_args.push_back({awesome_face_tex, raster_tex_conf});
+
+            rcbe::rendering::Material::MaterialConfig config {
+                std::move(shader_args),
+                std::move(texture_args)};
+            rcbe::rendering::Material material(
+                std::move(config));
+
+            rcbe::math::Transform trn {
+                rcbe::math::Matrix4x4 {
+                    1.0, 0., 0., 1.5,
+                    0., 0., 1., 0.,
+                    0., -1., 0., 0.,
+                    0., 0., 0., 1.}};
+
+            second_mesh.transform(trn);
+
+            wolf_object.addComponent<rcbe::geometry::Mesh>(std::move(second_mesh));
+            wolf_object.addComponent<rcbe::rendering::Material>(std::move(material));
+            wolf_object.addComponent<rcbe::visual::VisualTextureSet>(std::move(vts));
+        }
+
+        {
+            std::unordered_set<rcbe::visual::TexturePtr> tex_cache;
+            rcbe::rendering::Material::MaterialConfig mc {
+                {{"datamodel/data/rendering/shaders/viking_room.vert.spv",
+                  rdmn::render::ShaderType::vertex,
+                  rdmn::render::ShaderState::precompiled},
+                 {"datamodel/data/rendering/shaders/viking_room.frag.spv",
+                  rdmn::render::ShaderType::fragment,
+                  rdmn::render::ShaderState::precompiled}}};
+
+            rcbe::rendering::Material m(std::move(mc));
+            auto mat_ptr = std::make_shared<rcbe::core::CoreObject>(std::move(m));
+
+            for (size_t i = 0; i < viking_room_objects.size(); ++i) {
+                auto &c = viking_room_objects[i];
+                const auto &tex_component = c.getComponent<rdmn::render::RasterizerTexture>();
+                if (!tex_component)
+                    continue;
+                auto &rt = tex_component->as<rdmn::render::RasterizerTexture>();
+                const auto &vtex_ptr = rt.getVisualTexturePtr();
+
+                const auto it = tex_cache.find(vtex_ptr);
+                if (it == tex_cache.end()) {
+                    const auto [vtex_it, res] = tex_cache.insert(vtex_ptr);
+
+                    auto &mat = mat_ptr->as<rcbe::rendering::Material>();
+                    auto inserted_tex_it = mat.pushTexture(
+                        std::make_shared<rdmn::render::RasterizerTexture>(
```

Comment:
Just a comment: at this point the whole texture routine is overcomplicated, I suppose that `RasterizerTextures` should only be created inside `Renderer` class, and we should refactor whole thing to do so.

Reply:
Left a `TODO`